internal/api/stats_test.go

```diff
@@ -0,0 +1,109 @@
+package api
```

Comment:
wouldn't it be better if we name this package api_test.
Then we will have to call all exported items with 'api.' prefix as in api.Foo, like how this package will be used elsewhere.

we can be sure that our tests do not have any accidental dependency on unexported items.

please ignore my comment, if we are planning to write tests for unexported items in this file.

Reply:
found these pages

https://github.com/golang/go/issues/25223

first answer in https://stackoverflow.com/questions/19998250/proper-package-naming-for-testing-with-the-go-language/31443271#31443271

Reply:
That's a good idea. Will need to separate tests into two files because some time tests require internal variable.